.github/actions/build_4C/action.yml

```diff
@@ -0,0 +1,25 @@
+name: Build 4C
+description: Build a specified 4C target
+inputs:
+  build-directory:
+    description: Path to the build directory
+    required: true
+  build-targets:
+    description: Targets to build
+    required: true
+  log-file:
+    description: Path to the log-file
+    required: false
+    default: ${{ github.workspace }}/build.log
+runs:
+  using: composite
+  steps:
+    - name: Build 4C with targets ${{ inputs.build-targets }}
+      run: |
+        cd $BUILD_DIRECTORY
+        time cmake --build . --target $BUILD_TARGETS -- -j `nproc` 2>&1 | tee $LOG_FILE
```

Comment:
I think it is fine because when we set the shell to bash below. The option `pipefail` is added automatically. 
See `shell: bash --noprofile --norc -e -o pipefail {0}` from the workflow log.

But I agree that we might not even need this log. But we can keep it for now and remove it later.

Reply:
Let me remove it. We don't even need it for readthedocs.